WordPress/gutenberg
# Add cite to theme.json elements

## What?

Adds cite to theme.json elements. Partial for https://github.com/WordPress/gutenberg/issues/35735.

## Why?

To make it possible to style quote and pull quote citations separately.

## How?

The PR adds cite to theme.json elements and adds the new CSS class `wp-element-cite`.

- Updates the theme.json schema to include cite.
- Adds the wp-block-quote__citation and wp-block-pullquote__citation classes to the front so that the editor and front match.

_I am not sure if I updated the fixtures and tests correctly._
 
## Testing Instructions
Add the following to theme.json:
```
{
	"version": 2,
	"styles": {
		"elements": {
			"cite": {
				"color": {
					"text": "red"
				},
				"typography": {
					"fontSize": "40px",
					"lineHeight": ".7"
				}
			}
		}
	}
}
```

In either editor, add a quote block and a pullquote block.
Confirm that the styles applies correctly in the editors and front.

## Screenshots or screencast
